src/CredentialManager.js

```diff
@@ -0,0 +1,8 @@
+/* @flow */
+
+export interface ICredentialsManager<TCredentials> {
```

Comment:
this is sort of preference, but i'm used to naming my interfaces like `CredentialsManager` rather than `ICredentialsManager`

i mean, you already know it's an interface, so doing the pseudo-hungarian notation thing here always felt weird to me

Reply:
`CredentialsManager`?

Reply:
I'm fine with whatever, this was mostly a c# convention, I suppose it's not like one would ever mistake this for a real class